### PR TITLE
Added first version of bot

### PR DESCRIPTION
**Pull request by mrgrecha**

### CODE REVIEW

File: start_bot.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+require 'telegram/bot'
+require "openweather2"
+token = '449892555:AAHhCTRwYxf_za4GBfQibs7oYreocjS0um4'
+Openweather2.configure do |config|
+  config.endpoint = 'http://api.openweathermap.org/data/2.5/weather'
+  config.apikey = '000c898e740d056e93a25d9bf4cfca76'
```
Comment (anatoliliotych) [inline start_bot.rb]: move it to .env file and check dotenv gem

File: start_bot.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+require 'telegram/bot'
+require "openweather2"
+token = '449892555:AAHhCTRwYxf_za4GBfQibs7oYreocjS0um4'
```
Comment (anatoliliotych) [inline start_bot.rb]: it's not secure

File: start_bot.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+          length = text_from_message.split.length
+          current_city = text_from_message.split[1..length - 1].join(' ')
+          p current_city
+        end
+        bot.api.sendMessage(chat_id: message.chat.id, text: "Город изменился на #{current_city}")
+      when text_from_message.include?('погод') || text_from_message == '/get_weather'
+        some_object = Openweather2.get_weather(city: current_city, units: 'metric')
+        bot.api.sendMessage(chat_id: message.chat.id,
+        text:
+        "В городе #{some_object.city} ситация следующая:
```
Comment (anatoliliotych) [inline start_bot.rb]: typo :)

File: start_bot.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+require "openweather2"
+token = '449892555:AAHhCTRwYxf_za4GBfQibs7oYreocjS0um4'
+Openweather2.configure do |config|
+  config.endpoint = 'http://api.openweathermap.org/data/2.5/weather'
+  config.apikey = '000c898e740d056e93a25d9bf4cfca76'
+end
+Telegram::Bot::Client.run(token) do |bot|
+  current_city = 'Minsk'
+  bot.listen do |message|
+  text_from_message = message.text
```
Comment (anatoliliotych) [inline start_bot.rb]: check style guide

File: .env.rb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+require 'telegram/bot'
+require "openweather2"
+
```
Comment (anatoliliotych) [inline .env.rb]: remove empty line

File: start_bot.rb [hunk 6]
```diff
@@ -0,0 +1,34 @@
+require './.data'
```
Comment (anatoliliotych) [inline start_bot.rb]: what is data.rb? :)

File: start_bot.rb [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+require './.env'
+
+token = TELEGRAM_TOKEN
```
Comment (anatoliliotych) [inline start_bot.rb]: where do you store this token?

File: start_bot.rb [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+Openweather2.configure do |config|
+  config.endpoint = URL
+  config.apikey = WEATHER_TOKEN
```
Comment (anatoliliotych) [inline start_bot.rb]: where do you store this token?

File: start_bot.rb [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+end
+Telegram::Bot::Client.run(token) do |bot|
+  current_city = 'Minsk'
+    bot.listen do |message|
+      text_from_message = message.text
+        case
+        when text_from_message == '/start'
+            bot.api.sendMessage(chat_id: message.chat.id, text: "Hello, #{message.from.first_name}")
```
Comment (anatoliliotych) [inline start_bot.rb]: what about code style?

File: start_bot.rb [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+          if text_from_message.split.length > 1
+            length = text_from_message.split.length
+            current_city = text_from_message.split[1..length - 1].join(' ')
+            p current_city
+          end
+          bot.api.sendMessage(chat_id: message.chat.id, text: "Город изменился на #{current_city}")
+        when text_from_message.include?('погод') || text_from_message == '/get_weather'
+          some_object = Openweather2.get_weather(city: current_city, units: 'metric')
+          bot.api.sendMessage(chat_id: message.chat.id,
+          text:
```
Comment (anatoliliotych) [inline start_bot.rb]: check style!

File: start_bot.rb [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+            bot.api.sendMessage(chat_id: message.chat.id, text: "Hello, #{message.from.first_name}")
+        when text_from_message.include?('/city')
+          if text_from_message.split.length > 1
+            length = text_from_message.split.length
+            current_city = text_from_message.split[1..length - 1].join(' ')
+            p current_city
+          end
+          bot.api.sendMessage(chat_id: message.chat.id, text: "Город изменился на #{current_city}")
+        when text_from_message.include?('погод') || text_from_message == '/get_weather'
+          some_object = Openweather2.get_weather(city: current_city, units: 'metric')
```
Comment (anatoliliotych) [inline start_bot.rb]: use method to get weather so no need to store this here

File: start_bot.rb (hunk 6 above)
Comment (mrgrecha) [inline start_bot.rb]: It is a file with diffrent constants, that is in .gitignore. Is it a good practise to do in that way?

Comment (anatoliliotych) [inline start_bot.rb]: nope, use .env file with dotenv gem

File: start_bot.rb [hunk 12]
```diff
@@ -1,9 +1,11 @@
 require './.data'
```
Comment (anatoliliotych) [inline start_bot.rb]: do you use data.rb? :)

File: start_bot.rb (hunk 3 above)
Comment (anatoliliotych) [inline start_bot.rb]: ситуация

File: start_bot.rb [hunk 13]
```diff
@@ -1,39 +1,49 @@
...
+      else
+        bot.api.sendMessage(chat_id: message.chat.id, text:'I do not understand you')
       end
     end
 end
 
-def get_weather_object(current_city)
-  Openweather2.get_weather(city: current_city, units: 'metric')
-end
+
```
Comment (anatoliliotych) [inline start_bot.rb]: remove empty string

File: start_bot.rb [hunk 14]
```diff
@@ -1,39 +1,49 @@
...
       when text_from_message.include?('/city')
         if text_from_message.split.length > 1
           length = text_from_message.split.length
           current_city = text_from_message.split[1..length - 1].join(' ')
         end
         bot.api.sendMessage(chat_id: message.chat.id,
                             text: "Город изменился на #{current_city}")
-      when text_from_message.include?('погод')
+      when text_from_message.include?('погод') || text_from_message == '/get_weather'
         some_object = get_weather_object(current_city)
```
Comment (anatoliliotych) [inline start_bot.rb]: bad variable naming

File: start_bot.rb [hunk 15]
```diff
@@ -1,39 +1,49 @@
...
+require 'openweather2'
 Dotenv.load('.env')
 
 token = ENV['TELEGRAM_TOKEN']
 Openweather2.configure do |config|
   config.endpoint = ENV['URL']
   config.apikey = ENV['WEATHER_TOKEN']
 end
+
+def get_weather_object(current_city)
```
Comment (anatoliliotych) [inline start_bot.rb]: bad method naming

**Pull request merged**